<a href="https://colab.research.google.com/github/Payal2000/Customer-Segmentation-and-Clustering/blob/main/Customer_Segmenation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Problem Statement**

- Understanding the Target Customers for the marketing team to plan a strategy.

- We want to identify the most important shopping groups based on income, age, the mall shopping score.
- We would also want the ideal number of groups with a label for each.

# **Objective Market Segmentation**

- Divide the mall target market into approachable groups.
- Creating a subsets of market based on demographics behavioral criteria to better understand the target for marketing activities.


# **Our Approach**
1. Perform EDA ( Exploratoy Data Analysis )
2. Use KMEANS Clustering Algorithms to create segments
3. Use Summary Statistics on the clusters
4. Visualization

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt #Plotting
from sklearn.cluster import KMeans #Clustering
import seaborn as sns #Visualtization
import warnings
warnings.filterwarnings('ignore')

In [ ]:
file_url = "https://raw.githubusercontent.com/Payal2000/Customer-Segmentation-and-Clustering/main/data.csv"
data = pd.read_csv(file_url)


In [4]:
data.head(10)

CustomerID  Gender  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40
5           6  Female   22                  17                      76
6           7  Female   35                  18                       6
7           8  Female   23                  18                      94
8           9    Male   64                  19                       3
9          10  Female   30                  19                      72

## **Univariate Analysis**

In [5]:
data.describe()

CustomerID         Age  Annual Income (k$)  Spending Score (1-100)
count  200.000000  200.000000          200.000000              200.000000
mean   100.500000   38.850000           60.560000               50.200000
std     57.879185   13.969007           26.264721               25.823522
min      1.000000   18.000000           15.000000                1.000000
25%     50.750000   28.750000           41.500000               34.750000
50%    100.500000   36.000000           61.500000               50.000000
75%    150.250000   49.000000           78.000000               73.000000
max    200.000000   70.000000          137.000000               99.000000

### **Key statistical metrics**

- Count: There are 200 entries for CustomerID, Age, Annual Income (k$), and Spending Score (1-100), showing that the data is complete with no missing values.
- Mean ( Average value of each variable):
CustomerID: The mean is **100.5**, which is the midpoint ID given the range from 1 to 200.

- Age: The average age of the customers is approximately **38.85** years.

- Annual Income (k): The average annual income of customers is around **$60,560**.

- Spending Score (1-100): The average spending score, which indicates how much they spend (based on specific criteria), is **50.2**.